tests/tests/swfs/avm2/bytearray_writeobject/Test.as

```diff
@@ -0,0 +1,40 @@
+﻿package 
+{
+	import flash.utils.ByteArray;
+	public class Test
+	{
+		var TESTS = [
+		undefined,
+		null,
+		false,
+		true,
+		4,
+		4.5,
+		Infinity,
+		-Infinity,
+		NaN,
+		"test"
+		];
+		
+		public function testValue(value) {
+			var ba = new ByteArray();
+			ba.writeObject(value);
```

Comment:
note: would be nice to have some byte patterns checked too. Otherwise there's technically a possibility that we write wrong bytes that happen to roundtrip in Ruffle correctly.

Reply:
This is kind of hard, sometimes flash_lso generates different bytes than actual flash, but it's not wrong since they both mean the same thing. An example of this is when it serializes NaN, as there are multiple ways to represent that. Another example is arrays, since we don't support sparse arrays (which is serialized differently than a dense array).